homeassistant/components/overkiz/alarm_control_panel.py

```diff
@@ -0,0 +1,298 @@
+        supported_features=(
+            SUPPORT_ALARM_ARM_AWAY
+            | SUPPORT_ALARM_ARM_HOME
+            | SUPPORT_ALARM_ARM_NIGHT
+            | SUPPORT_ALARM_TRIGGER
+        ),
+        fn_state=_state_tsk_alarm_controller,
+        alarm_disarm=OverkizCommand.ALARM_OFF,
+        alarm_arm_home=OverkizCommand.SET_TARGET_ALARM_MODE,
+        alarm_arm_home_args=OverkizCommandParam.PARTIAL_1,
+        alarm_arm_night=OverkizCommand.SET_TARGET_ALARM_MODE,
+        alarm_arm_night_args=OverkizCommandParam.PARTIAL_2,
+        alarm_arm_away=OverkizCommand.SET_TARGET_ALARM_MODE,
+        alarm_arm_away_args=OverkizCommandParam.TOTAL,
+        alarm_trigger=OverkizCommand.ALARM_ON,
+    ),
+    # StatefulAlarmController
+    OverkizAlarmDescription(
+        key=UIWidget.STATEFUL_ALARM_CONTROLLER,
+        supported_features=(
+            SUPPORT_ALARM_ARM_AWAY | SUPPORT_ALARM_ARM_HOME | SUPPORT_ALARM_ARM_NIGHT
+        ),
+        fn_state=_state_stateful_alarm_controller,
+        alarm_disarm=OverkizCommand.DISARM,
+        alarm_arm_home=OverkizCommand.ALARM_ZONE_ON,
+        alarm_arm_home_args=[OverkizCommandParam.A],
+        alarm_arm_night=OverkizCommand.ALARM_ZONE_ON,
+        alarm_arm_night_args=[OverkizCommandParam.A, OverkizCommandParam.B],
+        alarm_arm_away=OverkizCommand.ALARM_ZONE_ON,
+        alarm_arm_away_args=[
+            OverkizCommandParam.A,
+            OverkizCommandParam.B,
+            OverkizCommandParam.C,
+        ],
+    ),
+    # MyFoxAlarmController
+    OverkizAlarmDescription(
+        key=UIWidget.MY_FOX_ALARM_CONTROLLER,
+        supported_features=SUPPORT_ALARM_ARM_AWAY | SUPPORT_ALARM_ARM_NIGHT,
+        fn_state=_state_myfox_alarm_controller,
+        alarm_disarm=OverkizCommand.DISARM,
+        alarm_arm_night=OverkizCommand.PARTIAL,
+        alarm_arm_away=OverkizCommand.ARM,
+    ),
+    # AlarmPanelController
+    OverkizAlarmDescription(
+        key=UIWidget.ALARM_PANEL_CONTROLLER,
+        supported_features=(
+            SUPPORT_ALARM_ARM_AWAY | SUPPORT_ALARM_ARM_HOME | SUPPORT_ALARM_ARM_NIGHT
+        ),
+        fn_state=_state_alarm_panel_controller,
+        alarm_disarm=OverkizCommand.DISARM,
+        alarm_arm_home=OverkizCommand.ARM_PARTIAL_DAY,
+        alarm_arm_night=OverkizCommand.ARM_PARTIAL_NIGHT,
+        alarm_arm_away=OverkizCommand.ARM,
+    ),
+]
+
+SUPPORTED_DEVICES = {description.key: description for description in ALARM_DESCRIPTIONS}
+
+
+async def async_setup_entry(
+    hass: HomeAssistant,
+    entry: ConfigEntry,
+    async_add_entities: AddEntitiesCallback,
+) -> None:
+    """Set up the Overkiz alarm control panel from a config entry."""
+    data: HomeAssistantOverkizData = hass.data[DOMAIN][entry.entry_id]
+    entities: list[OverkizAlarmControlPanel] = []
+
+    for device in data.platforms[Platform.ALARM_CONTROL_PANEL]:
+        if description := SUPPORTED_DEVICES.get(device.widget) or SUPPORTED_DEVICES.get(
```

Comment:
If `description` is None, you check if `device.ui_class` exists but you don't assign it to `description`, so it's going to be None. Is that on purpose ?

Reply:
Not sure if I understand what you mean. The aim here is to check if the EntityDescription key matches with the `device.widget` or the `device.ui_class`. If one of these is a match, it will use the current device + description with our `OverkizAlarmControlPanel`.